lib/python/treadmill_aws/hostmanager.py

```diff
@@ -15,22 +17,36 @@ def _instance_tags(hostname, role):
 
 
 def render_manifest(key_value_pairs):
-    """ Stub function to supply instance user_data during testing. """
+    """Returns formatted cloud-init from dictionary k:v pairs"""
 
     return "#cloud-config\n" + yaml.dump(
         key_value_pairs, default_flow_style=False, default_style='\'')
 
 
-def generate_hostname(domain, image):
-    """Generates hostname from role, domain and timestamp."""
+def generate_hostname(base, domain):
+    """Generates b32 string from current timestamp and returns formatted FQDN
+    """
     timestamp = str(time.time()).replace('.', '')
-    return '{}-{}.{}'.format(image.lower(), timestamp, domain)
+    b32time = aws.int2str(number=int(timestamp), base=32)
+    return '{}-{}.{}'.format(base.lower(), b32time, domain)
+
+
+def format_hostname(domain, hostname, role):
+    """Check if hostname defined and if it is a valid regex."""
+    if hostname:
```

Comment:
no need. 

just to hostname.format(time=xxx). if it has {time}, it will be replaced, if not, it will be ignored.